src/core/device.c

```diff
@@ -132,6 +299,18 @@ static int device_load(Unit *u) {
                         log_unit_debug_errno(u, r, "Failed to unescape name: %m");
         }
 
+        root_slice_unit = manager_get_unit(u->manager, SPECIAL_ROOT_SLICE);
```

Comment:
Maybe it would be better to check `verify_io_cost_supported()` here since it's required for both of these settings? Also I think typically we log, ignore the error, and return early if something is unsupported instead of failing hard. Something like:
```
if (!verify_io_cost_supported()) {
  // log and return message about being not supported but don't fail hard
}

// check for root slice
// call root_cgroup_apply_block_device_io_cost_qos and root_cgroup_apply_block_device_io_cost_model
```